web3/_utils/rpc_abi.py

```diff
@@ -57,6 +57,8 @@
     'personal_unlockAccount': ['address', None, None],
     'personal_sign': [None, 'address', None],
     'trace_call': TRACE_PARAMS_ABIS,
+    # parity
+    'parity_listStorageKeys': ['address', None, None, None],
```

Comment:
This seems fine for now, but in general, we probably want a different solution. It seems weird to have this included in this default middleware, for a non-default module. (Comment mostly intended for maintainers)

Reply:
Ah yes, that's familiar. I guess in that framework this is a method normalizer, really, since it depends on argument type and order.

Contract middleware is one I don't remember talking about. Do you have an example? Are you thinking of attaching a middleware to a specific contract instance, or across all, or what?

Reply:
> attaching a middleware to a specific contract instance

This, though thinking through it there isn't a *lot* that this gives us that couldn't be accomplished by pushing users towards using separate `w3` instances in cases where they want different behavior across different contract instances (think in-flight signing, or gas estimations).